jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -1233,4 +1234,82 @@ public interface Mount extends Closeable
          */
         Resource root() throws IOException;
     }
+
+    /**
+     * <p>Make a Resource containing a collection of other resources</p>
+     * @param resources multiple resources to combine as a single resource. Typically, they are directories.
+     * @return A Resource of multiple resources.
+     * @see ResourceCollection
+     */
+    public static ResourceCollection of(List<Resource> resources)
+    {
+        if (resources == null)
+            throw new IllegalArgumentException("No resources");
+
+        return new ResourceCollection(resources);
+    }
+
+    /**
+     * <p>Make a Resource containing a collection of other resources</p>
+     * @param resources multiple resources to combine as a single resource. Typically, they are directories.
+     * @return A Resource of multiple resources.
+     * @see ResourceCollection
+     */
+    public static ResourceCollection of(Resource... resources)
+    {
+        if (resources == null)
+            throw new IllegalArgumentException("No resources");
+
+        return new ResourceCollection(Arrays.asList(resources));
+    }
+
+    /**
+     * <p>Make a Resource containing a collection of other resources</p>
+     * @param resourceAsStrings multiple resources to combine as a single resource. Typically, they are directories.
+     * @return A Resource of multiple resources or null if no resources passed.
+     * @see ResourceCollection
+     * @throws IllegalArgumentException if a string cannot be converted to resource
+     */
+    public static ResourceCollection of(String... resourceAsStrings)
+    {
+        if (resourceAsStrings == null || resourceAsStrings.length == 0)
+            throw new IllegalArgumentException("No resources");
+
+        List<Resource> resources = new ArrayList<>();
+        for (String r : resourceAsStrings)
+        {
+            if (r == null)
+                continue;
+            r = r.trim();
+            if (StringUtil.isBlank(r))
+                continue;
+            try
+            {
+                resources.add(Resource.newResource(r));
```

Comment:
If the string is a URI reference `jar:file://foo.jar!/` from a user configuration, this won't work.

Reply:
Ah yes.  So I think the solution is to not have any String constructors to force the caller to do the resource conversion, and thus do the mounts if needed.
Working on it....

Reply:
I have this fixed in the original ResourceCollection PR, which has taken much of your changes in this PR
it's smaller than it was before you created this PR.